lib/awful/widget/tasklist.lua

```diff
@@ -461,16 +461,19 @@ function tasklist.new(args, filter, buttons, style, update_function, base_widget
         w:set_spacing(args.style and args.style.spacing or beautiful.tasklist_spacing)
     end
 
+    -- For the tests
+    function w._do_tasklist_update_now()
+        queued_update = false
+        if screen.valid then
+            tasklist_update(screen, w, args.buttons, args.filter, data, args.style, uf)
+        end
+    end
+
```

Comment:
For the commit message: I would replace "a reload" with "an update" (I think that refers to the third commit)